test/unit/application_test.rb

```diff
@@ -249,6 +249,27 @@ class ApplicationTest < ActiveSupport::TestCase
 
       assert_equal(expected.keys, app.latest_deploy_to_each_environment.keys)
     end
-  end
 
+    should "orders non-main environments so they show at the bottom of the table" do
+      Deployment.delete_all
+
+      app = FactoryBot.create(:application, name: SecureRandom.hex, repo: "alphagov/#{SecureRandom.hex}")
+
+      training = FactoryBot.create(:deployment, application: app, environment: "training")
+      preview = FactoryBot.create(:deployment, application: app, environment: "preview")
+      production = FactoryBot.create(:deployment, application: app, environment: "production")
+      staging = FactoryBot.create(:deployment, application: app, environment: "staging")
+      integration = FactoryBot.create(:deployment, application: app, environment: "integration")
+
+      expected = {
+        "integration" => integration,
+        "staging" => staging,
+        "production" => production,
+        "preview" => preview,
+        "training" => training,
```

Comment:
Ah, I understand why you have the dynamic sorting now: it's to accommodate the non-standard environments above.
I think we should just ignore them though. Looking at [Whitehall](https://release.publishing.service.gov.uk/applications/whitehall) for example (one of our older apps), its last Preview deployment was in 2015, and its last Training deployment was in 2019. We shouldn't be writing complex code just to support historical environments we're no longer deploying to.

I appreciate you did user research and asked the community their preference about these non-standard environments - perhaps you could do a follow-up Slack poll and ask devs if the envs should appear at the bottom of the list or be omitted entirely? (The latter wasn't an option in the first poll).

Reply:
Thank you for the suggestion, very useful, I've done the poll, and it came to light the non-main environments shouldn't show, so now the table on the app's pages only displays the main environments.

Reply:
Thanks - here's the [link to the poll](https://gds.slack.com/archives/CAB4Q3QBW/p1685961269387249) for future reference 😄